services/notebook/notebook.go

```diff
@@ -171,10 +172,12 @@ func (self *NotebookManager) UploadNotebookAttachment(ctx context.Context,
 
 func NewNotebookManager(
 	config_obj *config_proto.Config,
-	storage NotebookStore) *NotebookManager {
+	storage NotebookStore,
+	processor VqlProcessor) *NotebookManager {
```

Comment:
Can you suggest a use case for replacing the implementation with another implementation?

Reply:
removed the interface

Reply:
We want to replace the inline processor with one that will dispatch VQL tasks to a Kafka topic. Hiding it behind an interface allows us to make this change with minimal changes to the surrounding opensource code. Our footprint here will be very small but allow for the desired effect upstream

Reply:
This abstraction does not really fit here. I ask myself what is the function of the VqlProcessor? and it really is to convert the cell string to a result string (markup). This is basically the same function of the template processor so maybe this extra code can move there.

Generally to override the notebook service just override the whole service and update the UpdateNotebookCell() method. You will need to change that method anyway because it needs to work in a completely different way for sending to kafka, getting the results etc.

See this for example
https://github.com/Velocidex/cloudvelo/blob/28f247c56122eaa31c36b8853a78c905e3e35875/services/notebook/notebook.go#L11